vulture_os/toolkit/api_parser/catonetworks/catonetworks.py

```diff
@@ -0,0 +1,156 @@
+                continue
+            break
+        response.raise_for_status()
+        result = response.json()
+        if "errors" in result:
+            logger.error(f"[{__parser__}]:execute_query: Error while getting logs. reason : {response.content}", extra={'frontend': str(self.frontend)})
+            raise CatonetworksAPIError(f"Error while getting logs. reason : {response.content}")
+        return result
+
+    def get_logs(self):
+        query = '''{
+            eventsFeed(
+                accountIDs: [''' + self.catonetworks_account_id + ''']
+                marker: "''' + self.marker + '''"
+                filters: [''' + self.event_filter_string + "," + self.event_subfilter_string + ''']
+            ) {
+                marker
+                fetchedCount
+                accounts {
+                    id
+                    errorString
+                    records {
+                        time
+                        fieldsMap
+                    }
+                }
+            }
+        }'''
+        result = self.execute_query(query)
+        self.marker = result["data"]["eventsFeed"]["marker"]
+        self.fetched_count = int(result["data"]["eventsFeed"]["fetchedCount"])
+        return result["data"]["eventsFeed"]["accounts"][0]["records"]
+
+    def test(self):
+        try:
+            logs = self.get_logs()
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def format_log(self, log):
+        return json.dumps(log)
+
+    def execute(self):
+        try:
+            logs = self.get_logs()
+        except Exception as e:
+            logger.exception(f"[{__parser__}][get_logs]: {e}", extra={'frontend': str(self.frontend)})
+            raise CatonetworksAPIError(e)
+        # Downloading may take some while, so refresh token in Redis
+        self.update_lock()
+        if self.marker != "":
+            self.frontend.catonetworks_marker = self.marker
+            self.frontend.save(update_fields=["catonetworks_marker"])
+        self.write_to_file([self.format_log(log) for log in logs])
+        # Writting may take some while, so refresh token in Redis
+        self.update_lock()
+
+        while self.fetched_count==3000:
+
+            logs = self.get_logs()
+            # Downloading may take some while, so refresh token in Redis
+            self.update_lock()
+            if self.marker != "":
+                self.frontend.catonetworks_marker = self.marker
+                self.frontend.save(update_fields=["catonetworks_marker"])
+            self.write_to_file([self.format_log(log) for log in logs])
+            # Writting may take some while, so refresh token in Redis
+            self.update_lock()
```

Comment:
please combine your different blocks (first fetch and subsequent `while` loop) into a single `while` loop for clarity and brevity